org.epsg.openconfigurator/src/org/epsg/openconfigurator/model/PowerlinkRootNode.java

```diff
@@ -647,7 +648,19 @@ public Status importNodes(IFile projectFile,
 
                         InterfaceList.Interface intrfce = interfaceIterator
                                 .next();
-
+                        HeadNodeInterface headNodeInterface = null;
+                        List<HeadNodeInterface> headnodeInterfaceList = processingNode
+                                .getHeadNodeInterface();
+                        if (headnodeInterfaceList.size() != VALUE_ZERO) {
```

Comment:
redundant check. The for clause does not execute any way of the list is empty

Reply:
Yes, Considering only the for loop, this check might seem redundant. Considering future maintainability of the code, if there are additional statements added apart from the for loop alone, this check is necessary. We will change this to isEmpty() check.

Reply:
If you add additional code later you can add the check but now it just slows down the application and makes the code more complicated and difficult to understand. Please keep the changes as small as possible.